src/gluonts/mx/trainer/model_iteration_averaging.py

```diff
@@ -0,0 +1,272 @@
+
+        Returns
+        -------
+        """
+        # implement a naive strategy, use average_trigger as boolean
+        self.averaging_started = average_trigger
+        # raise NotImplementedError()
+
+    def apply(self, model: nn.HybridBlock) -> Optional[Dict]:
+        r"""
+        Parameters
+        ----------
+        model
+            The model of the current iteration.
+
+        Returns
+        -------
+        The averaged model, None if the averaging hasn't started.
+        """
+
+        if self.averaging_started:
+            self.update_average(model)
+
+        return self.averaged_model
+
+    def update_average(self, model: nn.HybridBlock):
+        r"""
+        Parameters
+        ----------
+        model
+            The model to update the average.
+        """
+        self.average_counter += 1
+        if self.averaged_model is None:
+            self.averaged_model = {
+                k: v.list_data()[0].copy()
+                for k, v in model.collect_params().items()
+            }
+        else:
+            alpha = 1.0 / self.average_counter
+            # moving average
+            for name, param_avg in self.averaged_model.items():
+                param_avg[:] += alpha * (
+                    model.collect_params()[name].list_data()[0] - param_avg
+                )
+
+    def load_averaged_model(self, model: nn.HybridBlock):
+        r"""
+        When validating/evaluating the averaged model in the half way of training, 
+        use load_averaged_model first to load the averaged model and overwrite the current model, 
+        do the evaluation, and then use load_cached_model to load the current model back.
+
+        Parameters
+        ----------
+        model
+            The model that the averaged model is loaded to.
+        """
+        if self.averaged_model is not None:
+            # cache the current model
+            if self.cached_model is None:
+                self.cached_model = {
+                    k: v.list_data()[0].copy()
+                    for k, v in model.collect_params().items()
+                }
+            else:
+                for name, param_cached in self.cached_model.items():
+                    param_cached[:] = model.collect_params()[name].list_data()[
+                        0
+                    ]
+            # load the averaged model
+            for name, param_avg in self.averaged_model.items():
+                model.collect_params()[name].set_data(param_avg)
+
+    def load_cached_model(self, model: nn.HybridBlock):
+        r"""
+        Parameters
+        ----------
+        model
+            The model that the cached model is loaded to.
+        """
+        if self.cached_model is not None:
+            # load the cached model
+            for name, param_cached in self.cached_model.items():
+                model.collect_params()[name].set_data(param_cached)
+
+
+class NTA_V1(IterationAveragingStrategy):
```

Comment:
That actually bothers me, too.
The problem is that, NTA_V1 and NTA_V2 are actually from the same algorithm in the same paper.
But the same paper has 2 versions, one on arxiv, another published on ICLR.
I guess only 1 of them is correct, but I cannot really tell which one is.
I did some experiments on some NLP tasks to test these 2 versions, but they show comparable results (in some cases V1 is better, and otherwise in other cases).
Currently, GluonNLP/MXNet is using V2, while Salesforce/PyTorch is using V1.